### PR TITLE
Use `TextareaAutosize` for the `textarea` resizing

### PR DESCRIPTION
**Pull request by kittaakos**
Aligned the `p-debounce` version between `@theia/debug` and `@theia/scm`

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

### CODE REVIEW

File: packages/scm/src/browser/scm-widget.tsx [hunk 1]
```diff
@@ -222,8 +221,10 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
                 tabIndex={1}
                 value={input.value}
                 onChange={this.setInputValue}
-                ref={this.setInput}>
-            </textarea>
+                innerRef={((ref: HTMLTextAreaElement) => this.input = ref)}
```
Comment (akosyakov) [inline packages/scm/src/browser/scm-widget.tsx]: we should not use arrow functions in react callbacks, it breaks react element caching

Comment (akosyakov) [reply packages/scm/src/browser/scm-widget.tsx]: please use `this.setInput` instead

File: yarn.lock [hunk 2]
```diff
@@ -7962,6 +7975,15 @@ rc@^1.0.1, rc@^1.1.6, rc@^1.2.1, rc@^1.2.7:
     minimist "^1.2.0"
     strip-json-comments "~2.0.1"
 
+react-autosize-textarea@^7.0.0:
```
Comment (akosyakov) [inline yarn.lock]: @marcdumais-work Could you check again please our deps? 🙏

File: packages/scm/src/browser/scm-widget.tsx [hunk 3]
```diff
@@ -245,37 +246,6 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
             this.input.focus();
         }
     }
-    /** TODO: a hack has to be implemented via React */
```
Comment (akosyakov) [inline packages/scm/src/browser/scm-widget.tsx]: very cool that you managed to replace it with one liner :)

**Comment by vince-fugnitto** [comment]
@kittaakos potentially fixes #3544 ? 😃

**Comment by akosyakov** [comment]
@vince-fugnitto no, but it was fixed in https://github.com/theia-ide/theia/pull/5416

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto no, but it was fixed in #5416 as well

Oh sorry, happy it's fixed! (or will be) :)

**Comment by kittaakos** [comment]
> please take care about arrow function

Done. Can we merge this into your branch, @akosyakov or we should wait for the CQ?

**Comment by akosyakov** [comment]
@kittaakos we don't need CQs for new prod deps if we follow: https://github.com/theia-ide/theia/wiki/Registering-CQs#wip---new-ecd-theia-intellectual-property-clearance-approach-experimental

I've just followed it and all new deps are of MIT license, so yes let's merge it. @marcdumais-work if you can double check later it would be nice anyway.

**Comment by kittaakos** [comment]
> so yes let's merge it.

OK. Ping me if something is odd with the `textarea` resizing.

File: yarn.lock (hunk 2 above)
Comment (marcdumais-work) [reply yarn.lock]: Done - I did the thorough process that includes recursive production dependencies, and all dependencies look good license-wise!

**Comment by marcdumais-work** [comment]
> @marcdumais-work if you can double check later it would be nice anyway.

Done - I confirm all dependencies, old and new, look good. Thanks for trying the new process by yourself, and feel free to do so again in the future - it's why I documented it. (ping me if any questions)

File: yarn.lock (hunk 2 above)
Comment (akosyakov) [reply yarn.lock]: I wonder how do you use ClearlyDefined UI, there are so many records. Do you scroll to find new deps or you use some filters?

Comment (marcdumais-work) [reply yarn.lock]: I do not really pay attention about new vs old dependencies; it takes a few minutes to go through them all, regardless. 

I sort by license, then I can go quickly with "page down", looking at the licenses for all packages. e.g. there are many many pages of MIT, so you can go though those really quick.

**Pull request merged**